core/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -151,9 +147,17 @@ public ExecutorScalingQueue() {
 
         @Override
         public boolean offer(E e) {
+            // first try to transfer to a waiting worker thread
             if (!tryTransfer(e)) {
+                // check if there might be spare capacity in the thread
+                // pool executor
```

Comment:
I'm not sure if we're just missing each other's point. I wasn't arguing for getting rid of `ExecutorScalingQueue`. I just think that a scaling threadpool should be always defined with min < max. This would mean that the condition `if (left > 0) {` can be removed and we end up with:

```
public boolean offer(E e) {
    return tryTransfer(e);
}
```